content/en/blog/_posts/2019-06-13-volume-cloning-alpha.md

```diff
@@ -0,0 +1,114 @@
+
+* name - name of the `PersistentVolumeClaim` object to use as source
+* kind - must be `PersistentVolumeClaim`
+* apiGroup - must be `""`
+
+The namespace of the source `PersistentVolumeClaim` (source) object is assumed to be the same as the namespace of the new `PersistentVolumeClaim` (dstination) being created object.
+
+```
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-clone
+  Namespace: demo-namespace
+spec:
+  storageClassName: csi-storageclass
+  dataSource:
+    name: src-pvc
+    kind: PersistentVolumeclaim 
+    apiGroup: ""
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+```
+
+When the `PersistentVolumeClaim` object is created, it will trigger provisioning of a new volume that is pre-populated with data from the specified `dataSource` volume.  It is the sole responsbility of the CSI Plugin to implement the cloning of volumes.
+
+## As a storage vendor, how do I add support for cloning to my CSI driver?
+
+To implement the cloning feature, a CSI driver MUST add cloning support to its controller capabilities `CLONE_VOLUME`. For details, see [the CSI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md).
+
+Cloning is an parameter added to the `CreateVolumeRequest` and checks to detect a clone option are added to the existing `Provision` function in the external provisioner.  From a CSI Plugin authors perspective, the requires that you modify your `CreateVolume` implementation to check and check the `CreateVolumeRequest` parameters for a data source.
+
+Although Kubernetes is as [minimally prescriptive](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/container-storage-interface.md#third-party-csi-volume-drivers) on the packaging and deployment of a CSI Volume Driver as possible, it provides a [suggested mechanism](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/container-storage-interface.md#recommended-mechanism-for-deploying-csi-drivers-on-kubernetes) for deploying an arbitrary containerized CSI driver on Kubernetes to simplify deployment of containerized CSI compatible volume drivers.
+
+
+The cloning features does not require any additional sidecar containers than the standard external provisioner.  There are however updates that have been made to the  external provisioner to enable cloning so you will need to update to the latest version of the provisioner <insert-link-or-version-number-here>
+
+## What are the limitations of alpha?
+
+The alpha implementation of cloning for Kubernetes has the following limitations:
+
+* Does not support cloning volumes across different namespaces
+* Does not support cloning volumes across different storage classes (backends)
+* Does not support cloning of `in-use` volumes
```

Comment:
Is this enforced? Or is this a "dont do it" warning?

Reply:
warning, don't do it; left to discretion of vendor so I'll remove this line